newdle/client/src/util/date.js

```diff
@@ -2,28 +2,56 @@ import moment, {HTML5_FMT} from 'moment';
 
 export const DEFAULT_TIME_FORMAT = 'HH:mm';
 
-export function toMoment(date, format = null) {
-  return date ? moment(date, format) : null;
+export function toMoment(date, format = null, timezone = null) {
+  return date ? moment.tz(date, format, timezone) : null;
 }
 
-export function serializeDate(date, format = HTML5_FMT.DATE) {
-  return moment(date).format(format);
+export function serializeDate(date, format = HTML5_FMT.DATE, timezone = null) {
+  return moment.tz(date, timezone).format(format);
 }
 
 export function overlaps([start1, end1], [start2, end2]) {
   return start1.isBefore(end2) && start2.isBefore(end1);
 }
 
 export function getHourSpan(input) {
-  const {timeSlots, defaultHourSpan, defaultMinHour, defaultMaxHour, duration, format} = input;
-  const timeSlotsMoment = timeSlots.map(c => toMoment(c, format));
+  const {
+    timeSlots,
+    busyTimes,
+    defaultHourSpan,
+    defaultMinHour,
+    defaultMaxHour,
+    duration,
+    format,
+    newdleTz,
+    userTz,
+  } = input;
+  let timeSlotsMoment = [];
+  if (userTz && newdleTz) {
```

Comment:
I don't think we can have any case with no `newdleTz` being set.

Actually, can `userTz` even be empty? I don't see anything about it in [the docs](https://momentjs.com/timezone/docs/#/using-timezones/guessing-user-timezone/).

So `userTz !== newdleTz` may be a better check here. But then I have the feeling that the logic here would simply not change anything, so maybe we don't need this check since we can always run the code here?

Reply:
I forgot to address this comment. The problem here is that this same method is used during creation, so it won't receive any userTz or newdleTz. I'll revisit this to see if I can have a cleaner logic for both the creation and the answer pages

Reply:
I noticed that I could just move the timezone for the outputted hours from the original `getHourSpan` method instead of modifying it